pubsub/topic.go

```diff
@@ -337,11 +350,11 @@ func (t *Topic) updateRequest(cfg TopicConfigToUpdate) *pb.UpdateTopicRequest {
 		pt.MessageStoragePolicy = messageStoragePolicyToProto(cfg.MessageStoragePolicy)
 		paths = append(paths, "message_storage_policy")
 	}
-	if cfg.RetentionDuration != 0 {
-		if cfg.RetentionDuration > 0 {
-			pt.MessageRetentionDuration = durationpb.New(cfg.RetentionDuration)
-		} else {
-			// If set to a negative value, clear MessageRetentionDuration.
+	if cfg.RetentionDuration != nil {
+		r := optional.ToDuration(cfg.RetentionDuration)
+		pt.MessageRetentionDuration = durationpb.New(r)
+		if r <= 0 {
```

Comment:
You've switched to `0` being meaning "clear the retention duration", are you sure? If so, please also update the field documentation (that right now just says "if negative"

Reply:
Oops, this was actually the case before too. I kept the behavior as "just negative".